internal/pkg/levee/levee.go

```diff
@@ -57,17 +57,28 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	for fn, sources := range sourcesMap {
 		for _, b := range fn.Blocks {
 			for _, instr := range b.Instrs {
-				v, ok := instr.(*ssa.Call)
-				if !ok {
-					continue
-				}
+				switch v := instr.(type) {
 
-				callee := v.Call.StaticCallee()
-				switch {
-				case fieldPropagators.IsFieldPropagator(v):
-					propagations[v] = propagation.Dfs(v, conf, taggedFields)
-					sources = append(sources, source.New(v))
-				case callee != nil && conf.IsSink(utils.DecomposeFunction(callee)):
+				case *ssa.Call:
+					callee := v.Call.StaticCallee()
+					switch {
+					case fieldPropagators.IsFieldPropagator(v):
+						propagations[v] = propagation.Dfs(v, conf, taggedFields)
+						sources = append(sources, source.New(v))
+					case callee != nil && conf.IsSink(utils.DecomposeFunction(callee)):
+						for _, s := range sources {
+							prop := propagations[s.Node]
+							if prop.HasPathTo(instr.(ssa.Node)) && !prop.IsSanitizedAt(v) {
```

Comment:
This isn't directly related to your change, but it feels like the caller knows too much here. I think this should be just one call, e.g. `prop.IsTainted(node)`. `prop` can easily determine that on its own, and to me it sounds like it's `prop`'s responsibility anyway. We may not need `HasPathTo` or `IsSanitizedAt` anymore.

Reply:
(Just to be clear, I don't think this should be changed in this PR.)